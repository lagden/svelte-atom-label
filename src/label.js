/* istanbul ignore file */
import Label from './Label.svelte'

export {
	Label
}
