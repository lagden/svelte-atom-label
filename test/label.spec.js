/* globals describe, afterEach, test, expect */

import {cleanup, render} from '@testing-library/svelte'
import {Label} from '../src/label.js'
import Slot from './Slot.svelte'

describe('Label', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(Label)
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot props', () => {
		const {container} = render(Label, {
			props: {
				class: 'customCss',
				for: 'inputID',
				required: true,
				disabled: false,
				'data-test': 1
			}
		})
		expect(container).toMatchSnapshot()
	})

	test('should match snapshot with slot', () => {
		const {container} = render(Slot)
		expect(container).toMatchSnapshot()
	})
})
