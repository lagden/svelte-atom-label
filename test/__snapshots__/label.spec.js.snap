// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should match snapshot default 1`] = `
<body>
  <div>
    <label
      class="_atom_frm__label"
    />
  </div>
</body>
`;

exports[`Label should match snapshot props 1`] = `
<body>
  <div>
    <label
      class="_atom_frm__label customCss _atom_frm__label___required"
      data-test="1"
      disabled="false"
      for="inputID"
      required="true"
    />
  </div>
</body>
`;

exports[`Label should match snapshot with slot 1`] = `
<body>
  <div>
    <label
      class="_atom_frm__label _atom_frm__label___disabled"
      disabled="true"
      for="test"
      required="false"
    >
      Test 
      <b>
        bold
      </b>
    </label>
  </div>
</body>
`;
